File: problems-extend/2201-2300/2279_maximum-bags-with-full-capacity-of-rocks/solution.go
import (
	"sort"
)

func maximumBags(capacity []int, rocks []int, additionalRocks int) int {
	needs := make([]int64, len(capacity))
	for i := range capacity {
		needs[i] = int64(capacity[i]) - int64(rocks[i])
	}
	sort.Slice(needs, func(i, j int) bool { return needs[i] < needs[j] })
	remaining := int64(additionalRocks)
	full := 0
	for _, need := range needs {
		if need > remaining {
			break
		}
		remaining -= need
		full++
	}
	return full
}

File: problems-extend/2201-2300/2279_maximum-bags-with-full-capacity-of-rocks/solution.ts
function maximumBags(capacity: number[], rocks: number[], additionalRocks: number): number {
    const needs = capacity.map((c, i) => c - rocks[i]).sort((a, b) => a - b);
    let remaining = additionalRocks;
    let full = 0;
    for (const need of needs) {
        if (need > remaining) {
            break;
        }
        remaining -= need;
        full++;
    }
    return full;
}
